### ISSUE TITLE
Create votes CRUD repository

### ISSUE THREAD

**Issue by pjeziorowski** [appwish/voteservice#2]
## Requirements
- propose DB schema for storing votes - that one might me used -> DB schema: https://files.slack.com/files-pri/TRYTNS8SX-FS37HGUQK/screenshot_2020-01-08_at_23.34.57.png
- prepare repository that is able create, delete and read votes (especially it should be able to count votes of given item
- for now we stick with upvotes only - no downvotes

depends on #1

**Comment by jlpnunes** [comment]
I am not able to see the screenshot link above and I am also not able to see the full table schema of VOTE on [https://app.creately.com/diagram/ToXBd2y63z4/view] . I will start with the service structure following the structure that you already defined for WISH.

**Comment by pjeziorowski** [comment]
![image](https://user-images.githubusercontent.com/44570539/72673078-925e8f80-3a65-11ea-8c71-24de11317d1f.png)

**Comment by pjeziorowski** [comment]
UserId - id of user who gave his vote
ItemId - id of upvoted item
ItemType - this one will tell us if the upvoted item is a wish, a comment or whatever
created - date of creation

(content from the screenshot you cannot access :S)

**Comment by pjeziorowski** [comment]
You may try to copy wish service and adjust the code to store votes instead of wishes

**Comment by jlpnunes** [comment]
Sure, I will try that. I have a question related to the schema. Why don't you add an ENUM field with value UP / DOWN for the vote. What is the plan to have a table to represent only UP votes now ?

**Comment by pjeziorowski** [comment]
Well good point, we can go with both upvotes and downvotes from the start. I guess it won't add much complexity.

**Comment by pjeziorowski** [comment]
@jlpnunes are you working on this or can I move it to 'todo'?

**Comment by jlpnunes** [comment]
@pjeziorowski , I am working on that. I had to stop this week but I will resume on the weekend.

**Comment by pjeziorowski** [comment]
OK nice. If you do not have the basic app structure yet, I've created a branch in other service with basic stuff - https://github.com/appwish/profileservice/tree/feature%231.

You can just copy all its contents and replace 'profile' with 'votes' and it should be fine for project bootstrap.

**Comment by jlpnunes** [comment]
@pjeziorowski can you check if I have the correct access rights to create a branch?

**Comment by jlpnunes** [comment]
@pjeziorowski is ItemType an enum field?

**Comment by pjeziorowski** [comment]
@jlpnunes check now - you should be able to do it. 

Yes, it can be an enum.

**Comment by Parables** [comment]
An extra hand is Available to help @jlpnunes

**Comment by pjeziorowski** [comment]
As there was no activity for 25 days maybe @Parables could take this task over? @jlpnunes ?

**Comment by Parables** [comment]
> OK nice. If you do not have the basic app structure yet, I've created a branch in other service with basic stuff - https://github.com/appwish/profileservice/tree/feature%231.
> 
> You can just copy all its contents and replace 'profile' with 'votes' and it should be fine for project bootstrap.

 @pjeziorowski  Link is broken ...

**Comment by pjeziorowski** [comment]
> > OK nice. If you do not have the basic app structure yet, I've created a branch in other service with basic stuff - https://github.com/appwish/profileservice/tree/feature%231.
> > You can just copy all its contents and replace 'profile' with 'votes' and it should be fine for project bootstrap.
> 
> @pjeziorowski Link is broken ...

@Parables Right now it's even easier - I would just copy whole https://github.com/appwish/profileservice as it contains nothing, just the basic structure of application with connection to Postgres :) So it's all you need for basic app structure.

**Comment by pjeziorowski** [comment]
I'm gonna create the boilerplate in #1 as I see no activity here

**Comment by pjeziorowski** [comment]
Ah, it's already done - sorry @jlpnunes, I have not noticed you pushed the changes to master. I'm closing #1 then. Thanks for contribution :)

**Comment by pjeziorowski** [comment]
done +-
